core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -303,6 +304,9 @@
   /** This caches absent paths in the UFS. */
   private final UfsAbsentPathCache mUfsAbsentPathCache;
 
+  /** Maps block ID to the list of block locations in UFS. */
+  private final ConcurrentHashMap<Long, List<String>> mUfsBlockLocations;
```

Comment:
should we use a guava cache (https://github.com/google/guava/wiki/CachesExplained) in order to limit the size of the map? And it could refresh also.